crates/rustc_codegen_spirv/src/builder/mod.rs

```diff
@@ -98,8 +98,29 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         self.current_span.unwrap_or(DUMMY_SP)
     }
 
+    // Given an ID, check if it's defined by an OpAccessChain, and if it is, return its ptr/indices
+    fn find_access_chain(&self, id: spirv::Word) -> Option<(spirv::Word, Vec<spirv::Word>)> {
+        let emit = self.emit();
+        let module = emit.module_ref();
+        let func = &module.functions[emit.selected_function().unwrap()];
+        let ptr_def_inst = func.all_inst_iter().find(|inst| inst.result_id == Some(id));
```

Comment:
Nice usecase for a def-use graph during codegen itself, heh. In my WIP stuff it would look like this:
```rust
fn find_access_chain(&self, ptr_value: Value) -> Option<(Value, &[Value])> {
    if let Value::DataInstOutput(inst) = ptr_value {
        let ptr_def_inst = &self.func.data_insts[inst];
        if ptr_def_inst.kind == ... {
            return Some((ptr_def_inst.inputs[0], &ptr_def_inst.inputs[1..]));
        }
    }
    None
}
```